src/sentry/search/django/backend.py

```diff
@@ -136,12 +136,22 @@ def _build_queryset(self, project, query=None, status=None, tags=None,
             )
 
         if tags:
-            matches = self._tags_to_filter(project, tags)
-            if not matches:
-                return queryset.none()
-            queryset = queryset.filter(
-                id__in=matches,
-            )
+            filter_tags = {k: v for k, v in six.iteritems(tags) if k[0] != '-'}
+            exclude_tags = {k[1:]: v for k, v in six.iteritems(tags) if k[0] == '-'}
```

Comment:
This is a minor nitpick, but this double iteration of `tags` should probably just be a single loop that adds to `filter_tags` or `exclude_tags` based on the prefix. It'd probably make sense to also define a behavior if the tag is included as both an inclusion and exclusion filter. I guess it'd be excluded by default as it stands now?